src/core/ext/transport/chttp2/transport/frame_secure.cc

```diff
@@ -0,0 +1,76 @@
+#include "src/core/ext/transport/chttp2/transport/frame_secure.h"
+
+#include <cstddef>
+#include <cstdint>
+
+#include "absl/status/status.h"
+
+#include "src/core/ext/transport/chttp2/transport/internal.h"
+#include "src/core/ext/transport/chttp2/transport/legacy_frame.h"
+#include "src/core/lib/event_engine/extensions/transport_extension.h"
+#include "src/core/lib/event_engine/query_extensions.h"
+#include "src/core/lib/iomgr/event_engine_shims/endpoint.h"
+#include "src/core/lib/slice/slice.h"
+#include "src/core/lib/slice/slice_buffer.h"
+
+absl::Status grpc_chttp2_secure_frame_parser_parse(void* parser,
+                                                   grpc_chttp2_transport* t,
+                                                   grpc_chttp2_stream* s,
+                                                   const grpc_slice& slice,
+                                                   int is_last) {
+  // Ignore frames from non-EventEngine endpoints.
+  if (!grpc_event_engine::experimental::grpc_is_event_engine_endpoint(
+          t->ep.get())) {
+    return absl::OkStatus();
+  }
+
+  grpc_chttp2_secure_frame_parser* p =
+      static_cast<grpc_chttp2_secure_frame_parser*>(parser);
```

Comment:
prefer `grpc_core::DownCast` (it will do a type check in debug builds)

Reply:
void pointer so can't downcast.